sqlmesh/cli/example_project.py

```diff
@@ -23,30 +25,66 @@ def _gen_config(
     start: t.Optional[str],
     template: ProjectTemplate,
 ) -> str:
-    connection_settings = (
-        settings
-        or """      type: duckdb
+    if not settings:
+        connection_settings = """      type: duckdb
       database: db.db"""
-    )
+
+        doc_link = "# Visit https://sqlmesh.readthedocs.io/en/stable/integrations/engines{dialect_link} for more information on configuring the connection to your execution engine."
+
+        dialect_alias = "mssql" if dialect == "tsql" else dialect
+
+        dialect_link = ""
+        if dialect_alias in CONNECTION_CONFIG_TO_TYPE:
+            required_fields = []
+            non_required_fields = []
+
+            for name, field in CONNECTION_CONFIG_TO_TYPE[dialect_alias].model_fields.items():
+                field_name = field.alias or name
+                default_value = field.get_default()
+
+                if isinstance(default_value, Enum):
+                    default_value = default_value.value
+                elif not isinstance(default_value, (str, int, bool)):
```

Comment:
Should we include `float` here? Perhaps worth considering a constant: `PRIMITIVES = (str, int, bool, float)`.